File: iSeeCars.Business/Helpers/InputHelper.cs
using System.Globalization;
namespace iSeeCars.Business.Helpers
{
    public static class InputHelper
    {
        public static int ReadInt(string message, Func<int, bool> validate, string errorMessage)
        {
            while (true)
            {
                Console.WriteLine(message);
                string input = Console.ReadLine();

                if (int.TryParse(input, out int value) && validate(value))
                    return value;

                ConsoleHelper.PrintErrorMessage(errorMessage);
            }

        }
        public static double ReadDouble(string message, Func<double, bool> validate, string errorMessage)
        {
            while (true)
            {
                Console.WriteLine(message);
                string input = Console.ReadLine();

                input = input.Replace(",", ".");


                if (double.TryParse(input, NumberStyles.Any, CultureInfo.InvariantCulture, out double value))
                    if (validate == null || validate(value))
                        return value;

                ConsoleHelper.PrintErrorMessage(errorMessage);
            }
        }

        public static string ReadString(string message, Func<string, bool> validate, string errorMessage)
        {

            while (true)
            {
                Console.WriteLine(message);
                string input = Console.ReadLine();

                if (validate(input))
                    return input;


                ConsoleHelper.PrintErrorMessage(errorMessage);
            }

        }

        public static void ShowEnum<T>()
        {
            foreach (var item in Enum.GetValues(typeof(T)))
            {
                Console.WriteLine($"{(int)item}.{item}");
            }
        }
    }
}

File: iSeeCars.java/src/main/java/iseecars/helper/InputHelper.java
package iseecars.helper;
import static java.lang.System.out;
import java.util.Scanner;
import java.util.function.Predicate;

public class InputHelper {
    private static final Scanner scanner = new Scanner(System.in);
    public static int readInt(String message, Predicate<Integer> validate, String errorMessage){
        while (true){
            out.println(message);
            try{
                int value = Integer.parseInt(scanner.nextLine());
                if(validate.test(value))
                    return value;

            } catch (Exception ignored){}

            ConsoleHelper.printErrorMessage(errorMessage);
        }
    }

    public static double readDouble(String message, Predicate<Double> validate,String errorMessage){
        while (true){
            out.println(message);

            try{
                double value = Double.parseDouble(scanner.nextLine().replace(",","."));
                if(validate == null || validate.test(value))
                    return value;
            }
            catch(Exception ignored){}
            ConsoleHelper.printErrorMessage(errorMessage);
        }
    }

    public static String readString(String message,Predicate<String> validate,String errorMessage){
        while(true){
            out.println(message);
            String input = scanner.nextLine();

            if(validate.test(input))
                return input;

            ConsoleHelper.printErrorMessage(errorMessage);
        }
    }
}
